packages/contracts-bridge/contracts/test/TokenRegistry.t.sol

```diff
@@ -1,10 +1,548 @@
+        address repr = tokenRegistry.getRepresentationAddress(
+            fuzzedDomain,
+            fuzzedId
+        );
+        assertEq(repr, localAddress);
+    }
+
+    event TokenDeployed(
+        uint32 indexed domain,
+        bytes32 indexed id,
+        address indexed representation
+    );
+
+    function test_ensureLocalTokenDeploy() public {
+        uint32 newDomain = 13;
+        bytes32 newId = bytes32("hey yoou");
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        address calculated = computeCreateAddress(address(tokenRegistry), 2);
+        vm.expectEmit(true, true, true, false);
+        emit TokenDeployed(newDomain, newId, calculated);
+        vm.prank(tokenRegistry.owner());
+        address deployed = tokenRegistry.ensureLocalToken(newDomain, newId);
+        assertEq(deployed, calculated);
+    }
+
+    function test_ensureLocalTokenExisting() public {
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        vm.prank(tokenRegistry.owner());
+        address addr = tokenRegistry.ensureLocalToken(
+            remoteDomain,
+            remoteTokenRemoteAddress
+        );
+        assertEq(addr, remoteTokenLocalAddress);
+    }
+
+    function test_ensureLocalTokenOnlyOwner() public {
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        vm.expectRevert("Ownable: caller is not the owner");
+        tokenRegistry.ensureLocalToken(remoteDomain, remoteTokenRemoteAddress);
+    }
+
+    uint256 iterations;
+
+    function test_ensureLocalTokenDeployFuzzed(uint32 domain, bytes32 id)
+        public
+    {
+        if (domain == localDomain) {
+            vm.prank(tokenRegistry.owner());
+            assertEq(
+                tokenRegistry.ensureLocalToken(domain, id),
+                TypeCasts.bytes32ToAddress(id)
+            );
+            return;
+        }
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        address calculated = computeCreateAddress(
+            address(tokenRegistry),
+            2 + iterations
+        );
+        vm.expectEmit(true, true, true, false);
+        emit TokenDeployed(domain, id, calculated);
+        vm.prank(tokenRegistry.owner());
+        address deployed = tokenRegistry.ensureLocalToken(domain, id);
+        assertEq(deployed, calculated);
+        iterations++;
+    }
+
+    function test_ensureLocalTokenOnlyOwnerFuzzed(address user) public {
+        // It's the second contract that is been deployed by tokenRegistry
+        // It deploys a bridgeToken during setUp() of BridgeTest
+        vm.assume(user != tokenRegistry.owner());
```

Comment:
we probably don't need to fuzz test OZ Ownable

Reply:
if we test that the function is covered by `onlyOwner` that's probs good enough

Reply:
maybe in the future, we remove the OZ ownable. Generally, I like to generalise almost every concrete unit test to a property-based one